### DISCUSSION TITLE
Using Epoptes over the Internet

### DISCUSSION THREAD

**Discussion by dashohoxha** [epoptes/epoptes#115] [Q&A]
Epoptes was designed to be used in a LAN (in a classroom). But nowadays classrooms are distributed due to the social distancing imposed by the virus (each student has to stay at home and classes are done online). I wonder if it is possible somehow to use Epoptes over the Internet. Of course with limited functionality (for example no WOL etc.)

I have developed some scripts for sharing the desktop through VNC remotely, with the help of ssh-tunnels: http://dashohoxha.fs.al/remote-desktop-access-with-vnc-and-sshtunnels/
I am not sure if the same technique or idea can be applied in this case. Logically, the Epoptes manager (which runs on the server) seems like a VNC client, so I think that it should work. But I am not sure. Maybe #114 (VPN with wireguard) is another option to do the same thing.

If you think that it might work, then maybe I will start a project about it on this hackathon: https://euvsvirus.org/ (which runs this weekend).

**Comment by alkisg** [comment]
I've been using Epoptes over the Internet due to coronavirus without any issues.
I'm not sure what issues you're referring to.

Of course, to have full functionality, a simple port forwarding isn't enough,
so the epoptes server either needs a "DMZ" public IP (even dynamic is OK),
or the clients need to connect to the server via a VPN, like OpenVPN, wireguard etc.

**Comment by dashohoxha** [comment]
That's great. I just never tried it that way.

The problem that I had in mind is that without a public IP it is not so easy for the clients (the computers of the students) to access the server (the computer of the teacher).

How do you make it work in this case?

**Comment by alkisg** [comment]
I don't have any use case where the teacher doesn't have a public IP.
All home Internet connections do involve a public IP... it doesn't need to be static...

**Comment by dashohoxha** [comment]
I see. It would still be interesting to know how you get a dynamic public IP (unless this is provided by the ISP and there is nothing to be done by configuration).
In any case, I am leaving this issue open as a reminder for adding something to the docs that mentions this case.

**Comment by alkisg** [comment]
For free dyndns services, have a look there:
https://www.maketecheasier.com/best-dynamic-dns-providers/

I'm not planning to add any documentation for this, as it's not really specific to Epoptes, but I could setup a community wiki like for LTSP, if Epoptes users are interested in maintaining it.
